inotify.go

```diff
@@ -116,6 +136,21 @@ func (w *Watcher) Remove(name string) error {
 	return nil
 }
 
+// Called from Close(); remove all watches.
+func (w *Watcher) removeAll() int {
+	successful := 0
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	for name, watch := range w.watches {
+		success, _ := syscall.InotifyRmWatch(w.fd, watch.wd)
```

Comment:
We only use removeAll when we've already closed.

Ah, and we do need watches for the inotify handles. Sorry.